src/config/resolve-config-editorconfig.js

```diff
@@ -8,7 +9,16 @@ const editorConfigToPrettier = require("editorconfig-to-prettier");
 const findProjectRoot = require("find-project-root");
 
 const maybeParse = (filePath, config, parse) => {
-  const root = findProjectRoot(path.dirname(path.resolve(filePath)));
+  // findProjectRoot will throw an error if we pass a nonexistent directory to
+  // it, which is possible, for example, when the path is given via
+  // --stdin-filepath. So, first, traverse up until we find an existing
+  // directory.
+  let dirPath = path.dirname(path.resolve(filePath));
+  const fsRoot = path.parse(dirPath).root;
+  while (dirPath !== fsRoot && !fs.existsSync(dirPath)) {
```

Comment:
Why is the `while` loop needed? I’m not saying it isn’t, I’m just asking why :) For example, what would happen if we bailed out if the directory does not exist? Is it possible to add tests for this?

Reply:
Good question! The loop is just trying to look up the resolve directory path, in case an editorconfig is present at some depth. As for just bailing out, that would also be fine. It would have the downside of ignoring editorconfigs that might otherwise have been used, but I suppose if you're running prettier via `stdin`, it doesn't really make sense to expect it to use your cwd's editorconfig, right?

If that sounds reasonable, let me know, and I'll change this to just bail.

Reply:
Ah, thanks, now I get it.

So, in current `master`, if `dirname(stdin-filepath)` happens to exist, .editorconfig (if any, and regardless of how high up the tree) is applied, right? I think we should either _always_ apply .editorconfig for stdin, or _never._ Given that we already apply .editorconfig in `master`, I guess we should keep that behavior. And as such we should keep the `while` loop.

Do we have a test for this `while` loop? If so we’re all good. Otherwise, could you please add one?